### DISCUSSION TITLE
Not seeing the static Serializer object on generated class

### DISCUSSION THREAD

**Discussion by kwonus** [jamescourtney/FlatSharp#359] [Q&A] [answered]
I am not seeing the static Serializer object [using Version 7 of FlatSharp].
![image](https://user-images.githubusercontent.com/12437390/209621289-5aa191bf-a8e4-4774-8b93-19ab22c6de6c.png)
Here is my IDL:
```
namespace AVSDK.FlatBuffers;
table AVBook {
    num:            uint8   (key);
    chapter_cnt:    uint8;
    chapter_idx:    uint16;
    name:           string  (required);
    abbreviations: [string] (required);
}
table AVBookIndex {
    index: [AVBook] (required);
}
root_type AVBookIndex;
```
All of the examples/tutorials seem to show the static Serializer object on generated thingies. What am I doing wrong? (the other boiler-plate operations are working. I do not get complaints from the compiler until the line with the Serializer static:
```
namespace AVSDK.Migration
{
    using AVSDK;
    using AVSDK.FlatBuffers;
    using System;

    class ConsoleApp
    {
        static void Main()
        {
            var api = new AVXAPI();

            var bookIndex = new AVBookIndex() { Index = new List<AVBook>() } ;

            for (var b = 0; b < api.Books.Length; b++)
            {
                var bk = api.Books[b];
                var book = new AVBook() { Name = bk.name, Abbreviations = new List<string>() };
                book.ChapterCnt = bk.chapterCnt;
                book.ChapterIdx = bk.chapterIdx;
                if (bk.abbreviations != null)
                    foreach (var abbr in bk.abbreviations)
                        book.Abbreviations.Add(abbr);
                book.Num = bk.num;
                bookIndex.Index.Add(book);
            }
            var maxBytesNeeded = AVBookIndex.Serializer.GetMaxSize(bookIndex);
            byte[] buffer = new byte[maxBytesNeeded];
            AVBookIndex.Serializer.Serialize(buffer, bookIndex);
        }
    }
}
```
```
3>C:\src\Digital-AV\z-series\FB\Program.cs(27,46,27,56): error CS0117: 'AVBookIndex' does not contain a definition for 'Serializer'
3>C:\src\Digital-AV\z-series\FB\Program.cs(29,25,29,35): error CS0117: 'AVBookIndex' does not contain a definition for 'Serializer'
```

**Comment by jamescourtney** [comment, marked as answer]
You need to tell FlatSharp that you want a serializer:

```idl
attribute "fs_serializer";

namespace AVSDK.FlatBuffers;
table AVBook (fs_serializer) {
    num:            uint8   (key);
    chapter_cnt:    uint8;
    chapter_idx:    uint16;
    name:           string  (required);
    abbreviations: [string] (required);
}
table AVBookIndex {
    index: [AVBook] (required);
}
root_type AVBookIndex;
```

Please take a look at the list of possible [FBS annotations](https://github.com/jamescourtney/FlatSharp/wiki/FBS-Annotations).

**Reply by kwonus** [reply]
I added the line:
```
attribute "fs_serializer";
```
And I did a clean/rebuild-all; and it still fails with the same error.

**Reply by jamescourtney** [reply]
You missed a line in the example I gave. There are two components:
- Adding the `attribute "fs_serializer";` declaration
- Annotating your table with `fs_serializer`, which my example shows: `table AVBook (fs_serializer)`

This is covered reasonably well in both the [wiki](https://github.com/jamescourtney/FlatSharp/wiki/FBS-Annotations#fs_serializer) and the [samples project](https://github.com/jamescourtney/FlatSharp/tree/main/samples/Example00-Basics).

**Reply by kwonus** [reply]
Terribly sorry; Thanks for the help! I love the simplicity that FlatSharp provides!

**Reply by jamescourtney** [reply]
No worries! Glad you find the library useful. Apologies for being a bit abrupt earlier as well. Please don't hesitate to reach out if you run into any other issues!
